src/ConservationChecker.jl

```diff
@@ -0,0 +1,331 @@
+#=
+    ConservationChecker
+
+This module contains functions that check global conservation of energy and water (and momentum - TODO).  
+=#
+module ConservationChecker
+
+using ClimaCoupler.Utilities
+using ClimaCore: ClimaCore, Geometry, Meshes, Domains, Topologies, Spaces, Fields, InputOutput
+using ClimaCore.Utilities: half
+using ClimaComms
+using NCDatasets
+using ClimaCoreTempestRemap
+using Dates
+using JLD2
+using UnPack
+using Plots
+using ClimaAtmos: RRTMGPI
+
+export AbstractCheck, EnergyConservationCheck, WaterConservationCheck, check_conservation!, plot_global_conservation
+
+abstract type AbstractCheck end
+
+"""
+    EnergyConservationCheck{A} <: AbstractCheck
+
+Struct of type `AbstractCheck` containing global energy conservation logs.
+"""
+struct EnergyConservationCheck{A} <: AbstractCheck
+    ρe_tot_atmos::A
+    ρe_tot_land::A
+    ρe_tot_ocean::A
+    ρe_tot_seaice::A
+    toa_net_source::A
+    ice_base_source::A
+    friction_sink::A
+end
+
+"""
+    WaterConservationCheck{A} <: AbstractCheck
+
+Struct of type `AbstractCheck` containing global water mass conservation logs.
+"""
+struct WaterConservationCheck{A} <: AbstractCheck
+    ρq_tot_atmos::A
+    ρq_tot_ocean::A
+    ρq_tot_land::A
+    ρq_tot_seaice::A
+end
+
+"""
+    check_conservation!(coupler_sim::CoupledSimulation)
+
+itertes over all specified conservation checks.
+"""
+check_conservation!(coupler_sim::CoupledSimulation, get_slab_energy, get_land_energy) =
+    map(x -> check_conservation!(x, coupler_sim, get_slab_energy, get_land_energy), coupler_sim.conservation_checks)
+
+"""
+        check_conservation!(
+        cc::EnergyConservationCheck,
+        coupler_sim,
+        )
+
+computes the total energy ∫ ρe dV of the various components
+of the coupled simulations, and updates cc with the values.
+
+TODO: move `get_slab_energy` and `get_land_energy` to their respective sims upon optimization refactor
+"""
+function check_conservation!(
+    cc::EnergyConservationCheck,
+    coupler_sim::CoupledSimulation,
+    get_slab_energy,
+    get_land_energy,
+)
+    @unpack model_sims, surface_masks = coupler_sim
+    @unpack atmos_sim, land_sim, ocean_sim, ice_sim = model_sims
+    radiation = atmos_sim.integrator.p.radiation_model # TODO: take out of global scope in ClimaAtmos
+    boundary_space = coupler_sim.boundary_space # thin shell approx (boundary_space[z=0] = boundary_space[z_top])
+
+    FT = eltype(coupler_sim.surface_masks.land)
+
+    # save radiation source
+    if radiation != nothing
```

Comment:
We have [this](https://github.com/CliMA/ClimaAtmos.jl/blob/a881905404ee6c50babc14c70ee57a0ece055587/examples/hybrid/callbacks.jl#L172-L187) in ClimaAtmos which integrates fluxes at the boundaries. It might be useful when you clean up the code.

Reply:
That's great! We can update to this upon the next ClimaAtmos release.